iree/compiler/Translation/IREEVM.h

```diff
@@ -56,12 +56,23 @@ struct InputDialectOptions {
   Type type = Type::none;
 };
 
+// Options controlling high level optimizations.
+struct HighLevelOptimizationOptions {
```

Comment:
neato - in the future I can imagine flags here for tuning the eval as well (`-iree-const-eval-favor=size/speed/etc`).

hmm `HLO`? :P I can't really think of anything better off the top of my head.
I think if we wanted to separate transform stages like this then we'd want to prefix the flags too so we can disambiguate like `iree-hlo-const-eval` (or `iree-hl-const-eval` etc) - that way when we do const-eval at a lower level (which I think would be useful) we can have an `iree-ll-const-eval` as well. Could still have one to turn it on globally, but for the other kind of things I could imagine living in here I'd see them living at other layers as well.

Another way to do this would be to change this to OptimizationOptions and make the const-eval a level flag - like `-iree-const-eval=0` is disabled, `-iree-const-eval=1` is run at the head of the pipeline only, `-iree-const-eval=2` is run at head and after HAL, etc with the assumption being kind of like -O you generally just want a knob you can turn up. When it's not denoting where it lives in the pipeline it'd be easier to put more optimization passes in here - especially those that may span layers (something like data compression may have parts run early to evaluate compression options and insert the code that then const-eval uses to perform the compression/packing/etc).

Reply:
(this is fine if we can't think of anything better, just wanted to brainstorm a bit :)

Reply:
I'll probably go with "hl". "HLO" has already cookie licked that name pretty well.

Agreed on turning these into levels. Gonna need that. I'm going to proceed as-is for now and get this closer to working at all (and with some optimization tradeoffs) and will def clean up/brainstorm more before real.